### PR TITLE
Feature/impressions listener

### PR DESCRIPTION
**Pull request by mredolatti**

### CODE REVIEW

File: splitio/proxy/controllers.go [hunk 1]
```diff
@@ -194,23 +195,35 @@ func mySegments(c *gin.Context) {
...
-	machineName := c.Request.Header.Get("SplitSDKMachineName")
-	data, err := ioutil.ReadAll(c.Request.Body)
-	if err != nil {
-		log.Error.Println(err)
-		controllerLocalCounters.Increment("request.error")
-		c.JSON(http.StatusInternalServerError, nil)
-		return
+func postImpressionBulk(impressionListenerEnabled bool) gin.HandlerFunc {
+	return func(c *gin.Context) {
+		//func postBulkImpressions(c *gin.Context) {
```
Comment (sarrubia) [inline splitio/proxy/controllers.go]: Remove this line

File: splitio/proxy/router.go [hunk 2]
```diff
@@ -72,7 +73,7 @@ func Run(options *ProxyOptions) {
 		api.GET("/splitChanges", splitChanges)
 		api.GET("/segmentChanges/:name", segmentChanges)
 		api.GET("/mySegments/:key", mySegments)
-		api.POST("/testImpressions/bulk", postBulkImpressions)
+		api.POST("/testImpressions/bulk", postImpressionBulk(options.ImpressionListenerEnabled))
```
Comment (sarrubia) [inline splitio/proxy/router.go]: @mredolatti Are you sure that this funcion postImpressionBulk is getting the gin.Context at runtime??

Comment (mredolatti) [inline splitio/proxy/router.go]: not that function, but the one returned by postImpressionBulk.
It works as a closure to have access to options.ImpressionListenerEnabled

File: splitio/proxy/controllers.go (hunk 1 above)
Comment (mredolatti) [inline splitio/proxy/controllers.go]: done sir

**Pull request merged**